new/server-stats-endpoint.md

```diff
@@ -0,0 +1,115 @@
+---
+RFC: unassigned
+Title: Stats Endpoint for Chef Server
+Author: Jay Mundrawala <jmundrawala@chef.io>
+Status: Draft
+Type: <Standards Track, Informational, Process>
+<Replaces: RFCxxx>
+<Tracking:>
+<  - https://github.com/chef/chef/issues/X>
+---
+
+# Stats Endpoint for Chef Server
+
+Monitoring Chef Server is an involved task. There are several services that make
+up a Chef Server, all of which need to be monitored. These services each have their
+own set of metrics, and each has it's own way obtaining those metrics. The Chef Server
+component itself has 2 ways that provide a different set of metrics: pushing to statsd 
+and through Folsom, which can be set up to push to Graphite.
+
+This RFC introduces a stats endpoint to Chef Server which is capable of providing metrics
+for Chef Server and all its co-located services.
+
+
+## Motivation
+
+    As a Chef Server operator,
+    I want to have a consistent way to get important Chef Server metrics,
+    so that I can easily monitor the health of my Chef Server out of the box.
+
+    As a Chef Server operator,
+    I want to understand the metrics I'm given,
+    so that I can easily monitor the health of my Chef Server.
+
+
+## Specification
+
+This RFC adds a series of `/_stats/$service` endpoints to Chef Server. Each service
+colocated on the box will be provided an endpoint, and with respond to GET requests.
+The response format will be a text format which is specified by 
+[Prometheus 0.0.4](https://prometheus.io/docs/instrumenting/exposition_formats/#text-format-details).
+This gives us the ability to use a few metrics types specified below along with the ability to give
+a description for any metrics which are provided.
+
+### Supported Metric Primitives
+The endpoints can provide metrics of following types, as defined by [Prometheus](https://prometheus.io/docs/concepts/metric_types/):
+- Counter - Monotonic cumulative metric. For example, number of requests served.
+- Gauge - Numeric measurement at a given point in time. For example, number of inflight requests.
+- Histogram - Distributes values into buckets. For example, response times.
```

Comment:
@jaymzh The specifics about which metrics was left as an implementation detail. However, to ease your concerns, one of the goals was to replace the need for that script. The idea is we will start with that script and add metrics as needed.